model/task/db.go

```diff
@@ -2559,7 +2559,7 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 	}
 
 	// Filter on Build Variants matching on display name or variant name if it exists
-	if len(opts.Variants) > 0 {
+	if len(opts.Variants) > 0 && opts.Variants[0] != "" {
```

Comment:
Is it possible that an empty string would appear in a non-zero position in the list of task/variant filters? If so it may make more sense to filter out the empty strings instead.

Reply:
Good catch!